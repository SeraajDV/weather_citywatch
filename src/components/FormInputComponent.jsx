
const FormInputComponent = () => {
  return (
    <section className="form-input">
      <form className="needs-validation mt-3" id="form" novalidate>
        <div className="row g-3 mb-3 justify-content-center">
          <div className="col-sm-3">
            <input type="text" className="form-control border-3" id="todoTaskInput" placeholder="Name of City..." />
            <div className="invalid-feedback">To do Task is required</div>
          </div>
        </div>
      </form>
    </section>
  )
}

export default FormInputComponent
