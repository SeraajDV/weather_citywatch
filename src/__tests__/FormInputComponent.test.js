import renderer from "react-test-renderer";
import FormInputComponent from "../components/FormInputComponent.jsx"

it("renders FormInputComponent correctly", () => {
  const tree = renderer
    .create(<FormInputComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/* REFERENCES
  ===========> 
  * I learnt to make snapshots from https://jestjs.io/docs/snapshot-testing
  */