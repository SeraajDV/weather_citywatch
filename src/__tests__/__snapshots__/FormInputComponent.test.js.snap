// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FormInputComponent correctly 1`] = `
<section
  className="form-input"
>
  <form
    className="mt-3"
    id="form"
  >
    <div
      className="row g-3 mb-3 justify-content-center"
    >
      <div
        className="col-sm-3"
      >
        <input
          className="form-control border-3"
          id="cityInput"
          onChange={[Function]}
          placeholder="Name of City/Town..."
          type="text"
          value=""
        />
      </div>
    </div>
  </form>
</section>
`;
